src/Npgsql/TypeHandlers/CompositeHandlers/IsValueType.cs

```diff
@@ -0,0 +1,7 @@
+﻿namespace Npgsql.TypeHandlers.CompositeHandlers
+{
+    static class IsValueType<T>
+    {
+        public static readonly bool Value = typeof(T).IsValueType;
```

Comment:
@roji [wrote](https://github.com/npgsql/npgsql/pull/2402#discussion_r270114598):

> It would be good to check that there's an actual perf benefit to having this - maybe just putting `typeof(T).IsValueType` everywhere is already optimized by the JIT (if it's not it should be :)).
>
> Otherwise I'd just call the property `IsValueType`...

FYI tracked down the [discussion on intrinsic support](https://github.com/dotnet/coreclr/issues/23208).

Reply:
Thanks for tracking this down @austindrenski! At least it's on their radar...